layer/trace_layer.cpp

```diff
@@ -53,6 +53,58 @@ const VkLayerProperties kLayerProps = {
 
 const std::vector<VkExtensionProperties> kDeviceExtensionProps = { VkExtensionProperties{ "VK_EXT_tooling_info", 1 } };
 
+/// An alphabetical list of device extensions which we do not report upstream if
+/// other layers or ICDs expose them to us.
+const char* const kDeviceExtensionBlacklist[] = {
+    // Look problematic:
+    "VK_EXT_external_memory_dma_buf",
+    "VK_EXT_external_memory_host",
+    // No app can expect this to always be present and we wouldn't want them to make
+    // decisions based on low level PCI bus info about the capture device.
+    "VK_EXT_pci_bus_info",
+    // Looks problematic:
+    "VK_EXT_physical_device_drm",
+    // Look problematic:
+    "VK_KHR_external_memory_fd",
+    "VK_KHR_external_memory",
+    // Supporting the CPU moving around descriptor set data directly has too many
+    // perf/robustness tradeoffs to be worth it.
+    "VK_VALVE_descriptor_set_host_mapping",
+};
+
+static void remove_unsupported_extensions(std::vector<VkExtensionProperties>& extensionProps)
+{
+    std::sort(extensionProps.begin(),
```

Comment:
What I have done is slightly different. I removed what I guess you objected to which was the `.erase()` inside a loop, but I've kept the algorithm in-place, working on an array and a vector, without building a tree or hash table. I have the sort of the extensions at the top which is O(n logn), but expected linear time as the input is usually already sorted. Then I've documented the linear character (caveat I'm treating short string compares as constant time when talking about complexity) of the rest of the function.

Reply:
@bradgrantham-lunarg Let me know if this is resolved when you have time next week or if you want the set of strings.

Reply:
I pushed a simpler version based on `std::remove_if(std::find_if())`. Since the list of extensions to screen out is currently length 1, we have a ways to go before the linear search of it is going to be a problem.